### ISSUE TITLE
你好 能不能开源一下泰语最终cer:12.46的zipformer的模型配置参数？

### ISSUE THREAD

**Issue by alanshaoTT** [SpeechColab/GigaSpeech2#9]
根据gigaspeech2论文附录的模型配置我设置的训练config是这样的 --num-encoder-layers 2,2,4,5,4,2 \
        --feedforward-dims 512,768,1536,2048,1536,768 \
        --nhead 4,4,4,8,4,4 \
        --encoder-dims 192,256,512,768,512,256 \
        --attention-dims 192,256,512,768,512,256 \
        --encoder-unmasked-dims 192,192,256,320,256,192 \
        --zipformer-downsampling-factors 1,2,4,8,4,2 \
        --cnn-module-kernels 31,31,15,15,15,31 \
        --decoder-dim 512 \
        --joiner-dim 512
但是这个维度与huggingface上开源的epoch-12.pt模型维度并不一样 我无法使用这个来load epoch-12.pt

**Comment by yfyeung** [comment]
> 根据gigaspeech2论文附录的模型配置我设置的训练config是这样的 --num-encoder-layers 2,2,4,5,4,2 --feedforward-dims 512,768,1536,2048,1536,768 --nhead 4,4,4,8,4,4 --encoder-dims 192,256,512,768,512,256 --attention-dims 192,256,512,768,512,256 --encoder-unmasked-dims 192,192,256,320,256,192 --zipformer-downsampling-factors 1,2,4,8,4,2 --cnn-module-kernels 31,31,15,15,15,31 --decoder-dim 512 --joiner-dim 512 但是这个维度与huggingface上开源的epoch-12.pt模型维度并不一样 我无法使用这个来load epoch-12.pt

@alanshaoTT 您好，最后一个 iter 就是默认的 zipformer large 的参数，如果有具体报错方便贴出来吗
```shell
./zipformer/train.py \
  --world-size 8 \
  --max-duration 1000 \
  --num-epochs 999 \
  --start-epoch 13 \
  --use-fp16 1 \
  --exp-dir zipformer/exp \
  --num-encoder-layers 2,2,4,5,4,2 \
  --feedforward-dim 512,768,1536,2048,1536,768 \
  --encoder-dim 192,256,512,768,512,256 \
  --encoder-unmasked-dim 192,192,256,320,256,192
```

**Comment by alanshaoTT** [comment]
> > 根据gigaspeech2论文附录的模型配置我设置的训练config是这样的 --num-encoder-layers 2,2,4,5,4,2 --feedforward-dims 512,768,1536,2048,1536,768 --nhead 4,4,4,8,4,4 --encoder-dims 192,256,512,768,512,256 --attention-dims 192,256,512,768,512,256 --encoder-unmasked-dims 192,192,256,320,256,192 --zipformer-downsampling-factors 1,2,4,8,4,2 --cnn-module-kernels 31,31,15,15,15,31 --decoder-dim 512 --joiner-dim 512 但是这个维度与huggingface上开源的epoch-12.pt模型维度并不一样 我无法使用这个来load epoch-12.pt
> 
> @alanshaoTT 您好，最后一个 iter 就是默认的 zipformer large 的参数，如果有具体报错方便贴出来吗
> 
> ```shell
> ./zipformer/train.py \
>   --world-size 8 \
>   --max-duration 1000 \
>   --num-epochs 999 \
>   --start-epoch 13 \
>   --use-fp16 1 \
>   --exp-dir zipformer/exp \
>   --num-encoder-layers 2,2,4,5,4,2 \
>   --feedforward-dim 512,768,1536,2048,1536,768 \
>   --encoder-dim 192,256,512,768,512,256 \
>   --encoder-unmasked-dim 192,192,256,320,256,192
> ```

报错主要是模型的encoder的ffnmodule维度不匹配 应该是icefall代码修改过了 我改过代码之后解决了。请问方便告知gigaspeech2 thai的zipformer模型是使用icefall的哪个recipe来实验的嘛？另外我发现解码方式和avg参数对模型效果影响很大，是否 方便告知解码方式

**Comment by yfyeung** [comment]
> > > 根据gigaspeech2论文附录的模型配置我设置的训练config是这样的 --num-encoder-layers 2,2,4,5,4,2 --feedforward-dims 512,768,1536,2048,1536,768 --nhead 4,4,4,8,4,4 --encoder-dims 192,256,512,768,512,256 --attention-dims 192,256,512,768,512,256 --encoder-unmasked-dims 192,192,256,320,256,192 --zipformer-downsampling-factors 1,2,4,8,4,2 --cnn-module-kernels 31,31,15,15,15,31 --decoder-dim 512 --joiner-dim 512 但是这个维度与huggingface上开源的epoch-12.pt模型维度并不一样 我无法使用这个来load epoch-12.pt
> > 
> > 
> > @alanshaoTT 您好，最后一个 iter 就是默认的 zipformer large 的参数，如果有具体报错方便贴出来吗
> > ```shell
> > ./zipformer/train.py \
> >   --world-size 8 \
> >   --max-duration 1000 \
> >   --num-epochs 999 \
> >   --start-epoch 13 \
> >   --use-fp16 1 \
> >   --exp-dir zipformer/exp \
> >   --num-encoder-layers 2,2,4,5,4,2 \
> >   --feedforward-dim 512,768,1536,2048,1536,768 \
> >   --encoder-dim 192,256,512,768,512,256 \
> >   --encoder-unmasked-dim 192,192,256,320,256,192
> > ```
> 
> 报错主要是模型的encoder的ffnmodule维度不匹配 应该是icefall代码修改过了 我改过代码之后解决了。请问方便告知gigaspeech2 thai的zipformer模型是使用icefall的哪个recipe来实验的嘛？另外我发现解码方式和avg参数对模型效果影响很大，是否 方便告知解码方式

recipe 还没整理好，可以参考不太规范的 https://github.com/yfyeung/icefall/blob/gigaspeech2/egs/gigaspeech2/ASR/decode.sh
解码先用 greedy 在 dev 上找出最佳的 epoch 和 avg 组合（遍历搜索），然后用 modified_beam_search beam_size 4 解码 dev / test

**Comment by yfyeung** [comment]
> > > 根据gigaspeech2论文附录的模型配置我设置的训练config是这样的 --num-encoder-layers 2,2,4,5,4,2 --feedforward-dims 512,768,1536,2048,1536,768 --nhead 4,4,4,8,4,4 --encoder-dims 192,256,512,768,512,256 --attention-dims 192,256,512,768,512,256 --encoder-unmasked-dims 192,192,256,320,256,192 --zipformer-downsampling-factors 1,2,4,8,4,2 --cnn-module-kernels 31,31,15,15,15,31 --decoder-dim 512 --joiner-dim 512 但是这个维度与huggingface上开源的epoch-12.pt模型维度并不一样 我无法使用这个来load epoch-12.pt
> > 
> > 
> > @alanshaoTT 您好，最后一个 iter 就是默认的 zipformer large 的参数，如果有具体报错方便贴出来吗
> > ```shell
> > ./zipformer/train.py \
> >   --world-size 8 \
> >   --max-duration 1000 \
> >   --num-epochs 999 \
> >   --start-epoch 13 \
> >   --use-fp16 1 \
> >   --exp-dir zipformer/exp \
> >   --num-encoder-layers 2,2,4,5,4,2 \
> >   --feedforward-dim 512,768,1536,2048,1536,768 \
> >   --encoder-dim 192,256,512,768,512,256 \
> >   --encoder-unmasked-dim 192,192,256,320,256,192
> > ```
> 
> 报错主要是模型的encoder的ffnmodule维度不匹配 应该是icefall代码修改过了 我改过代码之后解决了。请问方便告知gigaspeech2 thai的zipformer模型是使用icefall的哪个recipe来实验的嘛？另外我发现解码方式和avg参数对模型效果影响很大，是否 方便告知解码方式

模型训练和推理跟 zipformer 其他的 recipe 是一样的，主要就 bpe / lr_epochs / num_epochs 有差异，论文里最后一个 iter 用的 Zipformer-L，前面的 iter 都是 Zipformer-M

**Comment by alanshaoTT** [comment]
> > > > 根据gigaspeech2论文附录的模型配置我设置的训练config是这样的 --num-encoder-layers 2,2,4,5,4,2 --feedforward-dims 512,768,1536,2048,1536,768 --nhead 4,4,4,8,4,4 --encoder-dims 192,256,512,768,512,256 --attention-dims 192,256,512,768,512,256 --encoder-unmasked-dims 192,192,256,320,256,192 --zipformer-downsampling-factors 1,2,4,8,4,2 --cnn-module-kernels 31,31,15,15,15,31 --decoder-dim 512 --joiner-dim 512 但是这个维度与huggingface上开源的epoch-12.pt模型维度并不一样 我无法使用这个来load epoch-12.pt
> > > 
> > > 
> > > @alanshaoTT 您好，最后一个 iter 就是默认的 zipformer large 的参数，如果有具体报错方便贴出来吗
> > > ```shell
> > > ./zipformer/train.py \
> > >   --world-size 8 \
> > >   --max-duration 1000 \
> > >   --num-epochs 999 \
> > >   --start-epoch 13 \
> > >   --use-fp16 1 \
> > >   --exp-dir zipformer/exp \
> > >   --num-encoder-layers 2,2,4,5,4,2 \
> > >   --feedforward-dim 512,768,1536,2048,1536,768 \
> > >   --encoder-dim 192,256,512,768,512,256 \
> > >   --encoder-unmasked-dim 192,192,256,320,256,192
> > > ```
> > 
> > 
> > 报错主要是模型的encoder的ffnmodule维度不匹配 应该是icefall代码修改过了 我改过代码之后解决了。请问方便告知gigaspeech2 thai的zipformer模型是使用icefall的哪个recipe来实验的嘛？另外我发现解码方式和avg参数对模型效果影响很大，是否 方便告知解码方式
> 
> recipe 还没整理好，可以参考不太规范的 https://github.com/yfyeung/icefall/blob/gigaspeech2/egs/gigaspeech2/ASR/decode.sh 解码先用 greedy 在 dev 上找出最佳的 epoch 和 avg 组合（遍历搜索），然后用 modified_beam_search beam_size 4 解码 dev / test

好的好的  太感谢您了！！
